src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
```

Comment:
Ensure that the `ArrayListIndexOutOfBoundsException` is defined in your project, as it is used in the `checkIndex` and `checkIndexForAdd` methods.